octavia-cli/octavia_cli/headers.py

```diff
@@ -0,0 +1,126 @@
+        self.is_valid = all([self.name, self.value])
+
+    def clean(self):
+        return ApplicationHeader(
+            name=self.name.strip(),
+            value=self.value.strip()
+        )
+
+
+def deduplicate_api_headers(application_headers: List[ApplicationHeader]) -> List[ApplicationHeader]:
+    sorted_headers = sorted(application_headers, key=lambda header: header.name)
+    groups = groupby(sorted_headers, key=lambda header: header.name)
+    unique_headers = [
+        ApplicationHeader(
+            name=header_name,
+            value=sorted([header.value for header in group_headers])[0]
+        ) for header_name, group_headers in groups]
+    return unique_headers
+
+
+def read_yaml_header_configuration(header_configuration_path: str) -> List[ApplicationHeader]:
+    with open(header_configuration_path) as file:
+        content = yaml.load(file, EnvVarLoader)
+
+    headers = content.get("headers", None)
+
+    if not headers:
+        raise AttributeError("""
+        Please provide valid configuration file, like below
+        headers:
+          - name: Authorization
+            value: Basic dXNlcjpwYXNzd29yZA==
+          - name: Content-Type
+            value: application/json
+        """)
+
+    application_headers = [
+        ApplicationHeader(header_pair.get("name", None), header_pair.get("value")) for header_pair in headers
+    ]
+
+    valid_headers = [
+        app_header.clean() for app_header in application_headers if app_header.is_valid
+    ]
+
+    return deduplicate_api_headers(valid_headers)
+
+
+def get_argument_based_headers(api_headers: List[Tuple[str, str]]) -> List[ApplicationHeader]:
+    valid_api_headers = [
+        ApplicationHeader(
+            name=api_header_name,
+            value=api_header_value
+        ).clean() for api_header_name, api_header_value in api_headers
+    ]
+
+    return deduplicate_api_headers(valid_api_headers)
+
+
+def coalesce_headers(left: Optional[ApplicationHeader], right: Optional[ApplicationHeader]) -> Optional[ApplicationHeader]:
+    return left if left else right
+
+
+def merge_api_headers(
+        api_headers_arguments: List[ApplicationHeader],
+        file_based_list_of_headers: List[ApplicationHeader]) -> List[ApplicationHeader]:
+    application_headers_reference = {header.name: header for header in file_based_list_of_headers}
+
+    resolved_headers = []
+
+    for argument_based_header in api_headers_arguments:
+        file_based_header = None
+        if argument_based_header.name in application_headers_reference.keys():
+            file_based_header = application_headers_reference.pop(argument_based_header.name)
+
+        resolved_headers.append(
+            coalesce_headers(file_based_header, argument_based_header)
+        )
+
+    return resolved_headers + list(application_headers_reference.values())
+
+
+def validate_api_headers(api_headers: Optional[List[Tuple[str, str]]], api_headers_file: Optional[str]) -> List[ApplicationHeader]:
```

Comment:
```suggestion
def deserialize_api_headers(api_headers: Optional[List[Tuple[str, str]]], api_headers_file: Optional[str]) -> List[ApplicationHeader]:
```
I don't think any validation happens in this function.

Reply:
@Imbruced please let me know what you think about this suggestion.